### ISSUE TITLE
Support for audio and image files

### ISSUE THREAD

**Issue by elonen** [elonen/clapshot#59]
The easiest way to implement would be to have FFMPEG convert audio files into a waveform video, and images into still videos.

**Comment by elonen** [comment]
Maybe like this:
```
ffmpeg -i "$INPUTFILE" -r 60 -filter_complex "\
    color=c=white:s=2x720 [cursor]; \
    [0:a] showwavespic=s=1920x720:split_channels=1:draw=full, fps=60 [stillwave];\
    [0:a] showfreqs=mode=line:ascale=log:s=1920x180 [freqwave]; \
    [0:a] showwaves=size=1920x180:mode=p2p [livewave]; \
    [stillwave][cursor] overlay=(W*t)/(${DURATION}):0:shortest=1 [progress]; \
    [livewave][progress] vstack[stacked]; \
    [stacked][freqwave] vstack [out]; \
    " -map [out] -map 0:a \
    -c:v libx264 -c:a flac \
    -y output_progr.mp4
 ```
 
Omitting the [freqwave] layer makes this 2-3x faster, but spectral analysis might be quite useful for reviews & annotation.

**Comment by elonen** [comment]
More ideas. Detect media type by running a jq script on mediainfo JSON, and then convert into a video (like above) with an FFMPEG filter_complex chain.

```sql
CREATE TABLE media_types (
    type_name VARCHAR(255) NOT NULL PRIMARY KEY,
    precedence INTEGER NOT NULL,  -- run jq scripts in order from highest to lowest, pick the first that matches
    jq_script TEXT,   -- Stores the jq script for media type detection
    ffmpeg_script TEXT  -- Stores the FFMPEG script for converting media to video
);
```

**Video**
`jq '[.media.track[]["@type"]] | if (index("Video") != null) then "Video" else empty end'`

**Audio**
`jq '[.media.track[]["@type"]] | if (index("Audio") != null) and (index("Video") == null) then "Audio" else empty end'`

**Image**
`jq '[.media.track[]["@type"]] | if (index("Image") != null) and (index("Video") == null) and (index("Audio") == null) then "Image" else empty end'`

Use https://github.com/onelson/jq-rs to avoid shell invokes.

**Comment by elonen** [comment]
Implemented in v0.7.0
